app/models/request.rb

```diff
@@ -37,4 +37,11 @@ class Request < ApplicationRecord
       transitions from: [:pending, :approve, :paid], to: :rejected
     end
   end
+
+  def subtract_the_budget
+    @total_amount = Request.find_by(id: id).total_amount
+    @total_budget = Budget.first.total_budget
+    @result = @total_budget - @total_amount
+    Budget.first.update total_budget: @result
```

Comment:
sao lại Budget.first e
e chỉ có 1 budget dùng cho tất cả các giao dịch thôi à
sao k cho chọn budget để thanh toán ?

Reply:
> sao lại Budget.first e
> e chỉ có 1 budget dùng cho tất cả các giao dịch thôi à
> sao k cho chọn budget để thanh toán ?

dạ vâng, hiện tại thì e đã xóa hết mối quan hệ giữa budget và income, request, nên chỉ sử dụng 1 budget để giao dịch ạ. Như vậy thì bài toán có dễ quá không hay sao ạ ?

Reply:
dễ quá em nhá :v 
chọn buget để thanh toán đi e